auth/src/main/java/org/aerogear/auth/impl/UserPrincipalImpl.java

```diff
@@ -47,12 +45,12 @@
     private UserPrincipalImpl(final String username,
                               final ICredential credentials,
                               final String email,
-                              final Map<String, IRole> roles,
+                              final List<IRole> roles,
                               final AbstractAuthenticator authenticator) {
         super(authenticator);
         this.username = username;
         this.email = email;
-        this.roles = Collections.unmodifiableMap(new HashMap<String, IRole>(roles));
+        this.roles = roles;
```

Comment:
Even if we want to use a List instead of a Map, we should copy the List instead of simply assigning it.
If we simply assign, the caller still has a reference to the List and could change the List content corrupting the `UserPrincipalImpl` content.
The copy should be `unmodifiable`.